### PR TITLE
Add option to export spectral libraries through the command line, and remove "File > Export > ESP Features..." menu item 

### PR DESCRIPTION
**Pull request by henrytsanford**
This change adds the option to export spectral libraries through the command line. It creates a new group of arguments in the command line documentation for "Exporting other file types", which will eventually be expanded to include mProphet Features.

This change also removes the "File > Export > ESP Features" menu item.

### CODE REVIEW

File: pwiz_tools/Skyline/CommandArgUsage.resx [hunk 1]
```diff
@@ -928,4 +928,10 @@ greater peak area.</value>
   <data name="CommandArgs_ARG_FULL_SCAN_PRODUCT_ISOLATION_SCHEME_path_to_result_file_to_import_the_isolation_scheme" xml:space="preserve">
     <value>path/to/result-file to import the isolation scheme from</value>
   </data>
+  <data name="CommandArgs_GROUP_OTHER_FILE_TYPES" xml:space="preserve">
+    <value>Exporting other file types</value>
+  </data>
+  <data name="_exp_speclib_file" xml:space="preserve">
+    <value>Export a spectral library.</value>
```
Comment (brendanx67) [inline pwiz_tools/Skyline/CommandArgUsage.resx:935]: Let's make this: "Export experimental data in the current document to a spectral library file. Peak areas are used as spectrum intensities, peak apex times are used as retention times, and if the document has iRT values they will also be stored in the library."

File: pwiz_tools/Skyline/CommandLine.cs [hunk 2]
```diff
@@ -3106,6 +3115,42 @@ public bool ExportChromatograms(string chromatogramsFile, bool precursors, bool
             return true;
         }
 
+        /// <summary>
+        /// Export a spectral library (.blib) file from the document
+        /// </summary>
+        /// <param name="specLibFile"> File path to export the spectral library to</param>
+        /// <returns></returns>
```
Comment (brendanx67) [inline pwiz_tools/Skyline/CommandLine.cs:3122]: Add an explanation of the return value.

File: pwiz_tools/Skyline/CommandLine.cs [hunk 3]
```diff
@@ -3106,6 +3115,42 @@ public bool ExportChromatograms(string chromatogramsFile, bool precursors, bool
             return true;
         }
 
+        /// <summary>
+        /// Export a spectral library (.blib) file from the document
+        /// </summary>
+        /// <param name="specLibFile"> File path to export the spectral library to</param>
```
Comment (brendanx67) [inline pwiz_tools/Skyline/CommandLine.cs:3121]: Remove the leading space.

File: pwiz_tools/Skyline/CommandLine.cs [hunk 4]
```diff
@@ -3106,6 +3115,42 @@ public bool ExportChromatograms(string chromatogramsFile, bool precursors, bool
...
+        /// Export a spectral library (.blib) file from the document
+        /// </summary>
+        /// <param name="specLibFile"> File path to export the spectral library to</param>
+        /// <returns></returns>
+        public bool ExportSpecLib(string specLibFile)
+        {
+            _out.WriteLine(Resources.SkylineWindow_ShowExportSpectralLibraryDialog_Exporting_spectral_library__0____, specLibFile);
+            if (Document.MoleculeTransitionGroupCount == 0) // The document needs at least one peptide precursor
+            {
+                _out.WriteLine(Resources.CommandLine_ExportSpecLib_Error__The_document_must_contain_at_least_one_peptide_precursor_to_export_a_spectral_library);
```
Comment (brendanx67) [inline pwiz_tools/Skyline/CommandLine.cs:3128]: Does this get automatically changed to talk about one molecule if the document is a small molecule document? If so, mention that in the comment. If not, we need to fix that.

File: pwiz_tools/Skyline/Properties/Resources.resx [hunk 5]
```diff
@@ -13878,4 +13878,16 @@ If you choose Disable, you can enable Auto-select later with the "Refine &gt; Ad
   <data name="Server_GetKey___anonymous_" xml:space="preserve">
     <value> (anonymous)</value>
   </data>
+  <data name="CommandLine_ExportSpecLib_Failure_attempting_to_save_spectral_library_file__0_" xml:space="preserve">
+    <value>Failure attempting to save spectral library file {0}</value>
```
Comment (brendanx67) [inline pwiz_tools/Skyline/Properties/Resources.resx:13882]: Errors like this during command-line processing are generally expected to begin with "Error: "

**Review by brendanx67 — CHANGES_REQUESTED** [review]
A few comments, and I want you to add a completely new ZIP file for this test that is much smaller and your test should run in milliseconds.

File: pwiz_tools/Skyline/CommandLine.cs (hunk 4 above)
Comment (henrytsanford) [reply pwiz_tools/Skyline/CommandLine.cs]: I changed the error message from "peptide precursor" to simply "precursor" to address this, and updated the comment

**Review by brendanx67 — APPROVED** [review]

**Pull request merged**